fast-cart/src/components/Home.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import service from '../service'
+import {apiWrapper} from '../ulility-functions';
+import Loading from './Loading';
+import ProductGrid from './ProductGrid';
+import SideBar from './SideBar';
+
+const Home = () => {
+
+    const [productList, setProductList] = useState(undefined);
+
+    useEffect(()=>{
+        getAllProducts();
+    },[]);
+    const  getAllProducts = async () => {
+        let res =  await apiWrapper(service.getAllProducts,[]);
+        setProductList(res.products);
+    }
```

Comment:
I would suggest to follow an approach where you write all your API calls in a different place. 
You can also create a generic wrapper which returns you the state of any API call i.e., loading, error, data etc.